### PR TITLE
[P1] Admins/owner users should not be granted KP manage user access when using rostering on EH

### PR DESCRIPTION
**Pull request by mmeeoorroo**
# Trello card (URL):
https://trello.com/c/CgJOSONf

# Related PR(s):
https://github.com/Thinkei/employment-hero/pull/8149

# Summary
When request for an SSO URL do NOT add user to KP biz because it will grant full-access by default. Only add if necessary (using a flag).

# Resources:
- API for fetching biz access http://api.keypay.com.au/payroll/reference/business/business-access--get-all?v=latest
- Keypay UI for manage users/user access: https://ei.yourpayroll.com.au/Business/7053/RelatedUser

### CODE REVIEW

File: spec/access_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+        expect { service.business_access_granted?('', permissions: ['test'], any_matching: true) }
+            .to raise_error(ArgumentError, expected_message)
+        expect { service.business_access_granted?(nil, permissions: 'test', any_matching: true) }
+            .to raise_error(ArgumentError, expected_message)
+      end
+    end
+
+    context 'when pass an unsupported permission' do
+      let(:invalid_permissions) { %w(ShutdownKeypay BuyKeypay).sort }
```
Comment (khoido) [inline spec/access_service_spec.rb]: Nice permissions :trollface:

File: lib/keypay/sso_service.rb [hunk 2]
```diff
@@ -95,10 +89,10 @@ def create_keypay_user
     end
 
     def singlesignon_url
-      if eh_member['is_employer_or_owner']
-        "/business/#{kp_context[:payroll_organisation_id]}/singlesignon"
-      else
+      if kp_context[:payroll_id].present?
```
Comment (khoido) [inline lib/keypay/sso_service.rb]: I don't understand why we define the sso_url based on payroll_id ?

Comment (mmeeoorroo) [reply lib/keypay/sso_service.rb]: We allow non-KP-linked members to see rostering page. Those members don't have external_id (payroll_id).

Comment (mmeeoorroo) [reply lib/keypay/sso_service.rb]: If there is any KP user, with the same email with EH user, and the user is granted Manage or View Rosters permissions, we will still allow the user to see the roster page.

Comment (khoido) [reply lib/keypay/sso_service.rb]: Got it. Just a small note: you have a method call `has_payroll_id ?`, should use it here :trollface:

Comment (mmeeoorroo) [reply lib/keypay/sso_service.rb]: ahhh I missed that place

File: lib/keypay/keypay_url_helper.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      end
+
+      def ui_sso_rostering_url(email:, eh_context:, kp_context:)
+        ui_rostering_url = ui_rostering_url(email: email, kp_context: kp_context)
+        options = { url: ui_rostering_url }
+        ::KeypayIntegration::SsoService.new(eh_context, kp_context, true).keypay_sso_url(options)
+      end
+    end
+  end
+end
```
Comment (thukim) [inline lib/keypay/keypay_url_helper.rb]: missing new line

**Pull request merged**